### ISSUE TITLE
gixsql: lob related syntax issues

### ISSUE THREAD

**Issue by GitMensch** [mridoni/gixsql#3]
```
          EXEC SQL AT :DBS ALLOCATE :BLOBFLD END-EXEC.
          EXEC SQL AT :DBS
                LOB WRITE :LEN FROM :BUFFER
                INTO :BLOBFLD AT :OFFSET
           END-EXEC.
           EXEC SQL AT :DBS FREE :BLOBFLD END-EXEC
```

> syntax error, unexpected HOSTTOKEN, expecting SELECT or TOKEN or UPDATE or FOR
> error: syntax error, unexpected FROM
> syntax error, unexpected HOSTTOKEN, expecting SELECT or TOKEN or UPDATE or FOR

That is not critical for now, also because I don't know if that's correct syntax (I assume so), but if you see an option to adjust the COBOL code or the preparser, please drop a note.

**Comment by GitMensch** [comment]
Rechecked with 1.0.14, the lob statements _with_ `INTO` still don't generate, returning the same error message as before

> error: syntax error, unexpected SELECT, expecting DECLARE

To be explicit as we have many code in here - the following does not work , this is the code:

```cobol
           EXEC SQL AT :DBS
               SELECT lo_open(BLOBFLD,393216) INTO :DESCRIPTOR
                      FROM TAB WHERE TABKEY = :TABKEY;
           END-EXEC.

           EXEC SQL AT :DBS
               SELECT lo_lseek (:DESCRIPTOR, :OFFSET,0) INTO :RESINT
                      FROM TAB WHERE TABKEY = :TABKEY;
           END-EXEC.

           EXEC SQL AT :DBS
               SELECT LENGTH INTO :LEN FROM TAB
                      WHERE TABKEY= :TABKEY
           END-EXEC.

           EXEC SQL AT :DBS
               SELECT lo_close (:DESCRIPTOR) INTO :RESINT
                      FROM TAB WHERE TABKEY = :TABKEY;
           END-EXEC.
```

**Comment by mridoni** [comment]
Works with the current code in my test cases (including this one above), I will check if anything went wrong when packaging.

**Comment by GitMensch** [comment]
I don't think so, the last point was "there is likely a parser issue with the context" - but I did not found a way to "debug the parser/lexer states".

Nonetheless this is an actual show stopper, if nothing helps then I need to try to create a minimal example where this happens.

**Comment by mridoni** [comment]
Ok, this was a quick look, in a few days I should be able to resume working on it.

**Comment by mridoni** [comment]
I tried again with a cleanly-compiled version from the 1.0.14 sources on GitHub (the .tar.gz in Releases). There is some spurious screen output (probably some debug statement erroneously left in) but the translation works without errors (I have no way of testing code execution, obviously). I am pasting here the program I used as a (working) test case. When you have time, a minimal (non-working) test case would be appreciated so I can look at it sometime after this weekend.

Thanks

       IDENTIFICATION DIVISION.
       
       PROGRAM-ID. TSQL014C. 
       
       
       ENVIRONMENT DIVISION. 
       
       CONFIGURATION SECTION. 
       SOURCE-COMPUTER. IBM-AT. 
       OBJECT-COMPUTER. IBM-AT. 
       
       INPUT-OUTPUT SECTION. 
       FILE-CONTROL. 
       
       DATA DIVISION.  

       FILE SECTION.
      
       WORKING-STORAGE SECTION. 

           01 DATASRC     PIC X(64).
           01 DBS         PIC X(64).
           01 DBUSR       PIC X(64).
           01 DBPWD       PIC X(64).
           
           01 TABKEY                   PIC  9(018).
           01 LEN                      PIC S9(09).
           01 DESCRIPTOR               PIC S9(09).
           01 RESINT                   PIC S9(09).
           01 OFFSET                   PIC S9(09) VALUE 1.
       
       EXEC SQL 
            INCLUDE SQLCA 
       END-EXEC. 
       
       PROCEDURE DIVISION. 
 
       000-CONNECT.
         DISPLAY "DATASRC" UPON ENVIRONMENT-NAME.
         ACCEPT DATASRC FROM ENVIRONMENT-VALUE.
         DISPLAY "DATASRC_USR" UPON ENVIRONMENT-NAME.
         ACCEPT DBUSR FROM ENVIRONMENT-VALUE.
         DISPLAY "DATASRC_PWD" UPON ENVIRONMENT-NAME.
         ACCEPT DBPWD FROM ENVIRONMENT-VALUE.

         EXEC SQL
            CONNECT TO :DATASRC USER :DBUSR USING :DBPWD
         END-EXEC.      
         
         DISPLAY 'CONNECT SQLCODE: ' SQLCODE

         IF SQLCODE <> 0 THEN
            GO TO 100-EXIT
         END-IF.

       100-MAIN.

           EXEC SQL AT :DBS
               SELECT lo_open(BLOBFLD,393216) INTO :DESCRIPTOR
                      FROM TAB WHERE TABKEY = :TABKEY;
           END-EXEC.

           EXEC SQL AT :DBS
               SELECT lo_lseek (:DESCRIPTOR, :OFFSET,0) INTO :RESINT
                      FROM TAB WHERE TABKEY = :TABKEY;
           END-EXEC.

           EXEC SQL AT :DBS
               SELECT LENGTH INTO :LEN FROM TAB
                      WHERE TABKEY= :TABKEY
           END-EXEC.

           EXEC SQL AT :DBS
               SELECT lo_close (:DESCRIPTOR) INTO :RESINT
                      FROM TAB WHERE TABKEY = :TABKEY;
           END-EXEC.

       100-EXIT. 
             STOP RUN.

       200-END.

**Comment by GitMensch** [comment]
... that issue above is an old bug, which I've worked around before; when starting with fresh GixSQL the issue came back because there was no patch integrated. I'll have a quick look and possibly do a PR.

**Comment by GitMensch** [comment]
Closed as solved with 1.0.16, with copying the source sample above as-is (fixed-form reference-format with bad indent) there is no bad error any more but a clear

> error: Wrong file format or unexpected end of file

when using a correct fixed-form reference-format the preprocessing works, leaving only #2 which has its own issue.

**Comment by GitMensch** [comment]
This seems to haunt me:

> PROG.pco:1720: error: syntax error, unexpected SELECTFROM
PROG.pco:1751: error: syntax error, unexpected SELECTFROM
... and much more of that

```sql
          EXEC SQL
               UPDATE TAB SET DESCRIPTOR = lo_open(BLOB1,393216)
                      WHERE MID=:MID
           END-EXEC.
      *
           IF SQLCODE NOT = 0
              EXIT SECTION.
      * line 1721 follows
           EXEC SQL
              SELECT LENGTH FROM TAB INTO :LEN WHERE MID=:MID
           END-EXEC.
```

"of course" I've did a s/FILE/FLE/g after seeing that there are sections the end with `FILE` - but that did not help (it did several versions before).

Wasn't there some nice option to enable debug output, allowing us to see the (likely bad) parser state here?
